changelog: Clever Tools 3.5.2 (#253)
CleverCloud/documentation: content/changelog/2024-04-11-clever-tools-3.5.2.md

---
title: Clever Tools 3.5.2 is available with cURL, KV, Tasks and more!
date: 2024-04-11
tags:
  - clever-tools
  - cli
authors:
  - name: David Legrand
    link: https://github.com/davlgd
    image: https://github.com/davlgd.png?size=40
  - name: Hubert Sablonière
    link: https://github.com/hsablonniere
    image: https://github.com/hsablonniere.png?size=40
description: Lots of new features included!
excludeSearch: true
---

> Clever Tools 3.5 is now available, and this branch brings lots of new important features. First, `clever curl` is now public and listed as an official command. It helps you to send cURL requests with the auth context of your Clever Tools configuration. Thus, you can simply use our API v2 or v4. Their documentation pages are available [here](/api/).

Reviewer: ```suggestion
Clever Tools 3.5 is now available, and this branch brings lots of new important features. First, `clever curl` is now public and listed as an official command. It helps you to send cURL requests with the auth context of your Clever Tools configuration. Thus, you can use Clever Cloud [API v2 or v4](/api/)..
```

JSON format is supported for more commands, the `--since` option now supports a duration value. For example if you want to get logs since 2 hours ago, you can use `clever logs --since 2h`. This command documentation is available [there](https://github.com/CleverCloud/clever-tools/blob/master/docs/applications-deployment-lifecycle.md#logs).

> Clever Tasks can now be easily created and configured from Clever Tools. They are applications which can be run on demand, not awaiting any HTTP request on the `8080` port, but needing a `CC_RUN_COMMAND` to execute. Once it's done, the application stops. You're just billed for the execution time. It can help you to make some checks, compilation, file conversions, etc.

Reviewer: ```suggestion
Clever Tasks can now be directly created and configured from Clever Tools. They're applications which can be run on demand, not awaiting any HTTP request on the `8080` port, but needing a `CC_RUN_COMMAND` to execute. Once it's done, the application stops. You're just billed for the execution time. It can help you to make some checks, compilation, file conversions, etc.
```

> To create a Clever Task using Python for example, just:

Reviewer: ```suggestion
To create a Clever Task using Python for example:
```

```
# We create a Python App and its Git repository
mkdir pythonTask && cd pythonTask
echo 'print("Hello, from a Clever Cloud Task!")' > task.py
git init && git add . && git commit -m "Initial commit"

# We deploy this app as a Task
clever create -t python --task 'python task.py'
clever deploy
```

> If you want to check if an application will be deployed as a `TASK` or the `REGULAR` way, use `clever status`. Of course, you can change the executed command, with a simple environment variable. Just set another value, for example with a bash script:

Reviewer: ```suggestion
If you want to check if an application will be deployed as a `TASK` or the `REGULAR` way, use `clever status`. Change the executed command using an environment variable. Set another value, for example with a bash script:
```

```
clever env set CC_RUN_COMMAND "bash a_bash_script.sh"
```

> We'll enhance Clever Tasks other time, we already have ideas and a plan to do so in the coming months. Feel free to tell us about your needs or ideas on this subject!

Reviewer: ```suggestion
Clever Tasks is in current development for enhancements over the coming months. Feel free to tell us about your needs or ideas on this subject!
```

> Last but not least, we're introducing the access to MateriaDB KV. Those granted with alpha access can now create a serverless, synchronously-replicated (over our 3 datacenters in Paris) key-value add-on. You'll be able to use it with multiples kind of clients. First to be supported is Redis API. Thus, you can just:

Reviewer: ```suggestion
Last but not least, we're introducing the access to MateriaDB KV. Those granted with alpha access can now create a serverless, synchronously-replicated (over our 3 datacenters in Paris) key-value add-on. You'll be able to use it with multiples kind of clients. First to be supported is Redis API. Thus, you can run:
```
